File: fix_nesting.py
import re
import os

target_file = r'c:\Users\Lenovo\Desktop\wallet app\index.html'

with open(target_file, 'r', encoding='utf-8') as f:
    content = f.read()

# Regex to find the broken block between our specific cleanup comments
# It will remove everything between the tags, including the stray </div> at line 1299
pattern = re.compile(r'<!--\s\[CLEANUP START - 2026-04-02\]\s-->.*?<!--\s\[CLEANUP END - 2026-04-02\]\s-->', re.DOTALL)
replacement = '<!-- [FIX: Restored SPA Viewport Nesting by removing stray </div> and garbage lines - 2026-04-02] -->'

new_content = pattern.sub(replacement, content)

if new_content != content:
    with open(target_file, 'w', encoding='utf-8') as f:
        f.write(new_content)
    print("SUCCESS: Structural fix applied.")
else:
    print("ERROR: Could not find the cleanup block in the file.")

File: fix_nesting.js
const fs = require('fs');
const path = require('path');

const targetFile = 'c:\\Users\\Lenovo\\Desktop\\wallet app\\index.html';

try {
    let content = fs.readFileSync(targetFile, 'utf8');

    // Regex to find the broken block between our specific cleanup comments
    // It will remove everything between the tags, including the stray </div> at line 1299
    // Using a more flexible regex for whitespaces
    const pattern = /<!--\s+\[CLEANUP START - 2026-04-02\]\s+-->[\s\S]*?<!--\s+\[CLEANUP END - 2026-04-02\]\s+-->/g;
    const replacement = '<!-- [FIX: Restored SPA Viewport Nesting by removing stray </div> and garbage lines - 2026-04-02] -->';

    const newContent = content.replace(pattern, replacement);

    if (newContent !== content) {
        fs.writeFileSync(targetFile, newContent, 'utf8');
        console.log("SUCCESS: Structural fix applied via Node.js.");
    } else {
        console.log("ERROR: Could not find the cleanup block in the file via Node.js.");
    }
} catch (err) {
    console.error("FAILED to read or write file:", err);
}
